hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -466,6 +467,65 @@ object LowerTableIR {
         invoke("sum", TInt64,
           stage.mapCollect(relationalLetsAbove)(rows => foldIR(mapIR(rows)(row => Consume(row)), 0L)(_ + _)))
 
+      case TableToValueApply(child, TableCalculateNewPartitions(nPartitions)) =>
+        val stage = lower(child)
+        val sampleSize = math.min(nPartitions * 20, 1000000)
+        val samplesPerPartition = sampleSize / stage.numPartitions
+        val keyType = child.typ.keyType
+        val samplekey = AggSignature(TakeBy(),
+          FastIndexedSeq(TInt32),
+          FastIndexedSeq(keyType, TFloat64))
```

Comment:
We should be able to collect the sample faster using SeqSample instead of TakeBy, but that's not urgent. Also not sure the purpose of inserting the partition mins/maxs into the sample.